File: server/services/menuService.go
package services

import (
	"github.com/MatheusABA/restaurant-project/server/controller/menu/dto"
	"github.com/MatheusABA/restaurant-project/server/model"
	"github.com/MatheusABA/restaurant-project/server/repositories"
)

func CreateMenuItem(req dto.CreateMenuItemRequest) (*model.MenuItem, error) {
	item := model.MenuItem{
		Name:     req.Name,
		Price:    req.Price,
		Category: req.Category,
		IsActive: req.IsActive,
	}
	return repositories.CreateMenuItem(&item)
}

func GetAllMenuItems() ([]model.MenuItem, error) {
	return repositories.GetAllMenuItems()
}

File: web/src/api/menuService.ts
import api from "./api";

interface MenuItem {
  name: string;
  price: number;
  category: string;
  is_active: boolean
}


class MenuService {

  async createMenuItem(token: string, data: MenuItem) {
    const response = await api.post('/menu/createMenuItem', data, {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    });
    return response.data;
  }

  async getAllMenuItems(token: string) {
    const response = await api.get('/menu/getAllMenuItems', {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    return response.data;
  }



}

export default new MenuService();